sdk/nodejs/provider.ts

```diff
@@ -2222,7 +2247,14 @@ export namespace yaml {
     export class ConfigFile extends CollectionComponentResource {
         constructor(name: string, config?: ConfigFileOpts, opts?: pulumi.ComponentResourceOptions) {
             super("kubernetes:yaml:ConfigFile", name, config, opts);
-            const text = fs.readFileSync(config && config.file || name).toString();
+            const fileId = config && config.file || name
+            let text = "";
+            if (fileId.startsWith("http://") || fileId.startsWith("https://")) {
+                text = syncreq.default("GET", fileId).getBody().toString()
```

Comment:
I *think* we can pretty easily incorporate async IO here, which would be better if it doesn't negatively impact usabilty.  Specifically - can't you do something like this:

```
export class ConfigFile extends CollectionComponentResource {
        constructor(name: string, config?: ConfigFileOpts, opts?: pulumi.ComponentResourceOptions) {
            super("kubernetes:yaml:ConfigFile", name, config, opts);
            const textP = fs.readFilePromise(config && config.file || name);
            const objsP = textP.then(text => jsyaml.safeLoadAll(text.toString()));
            this.resources = pulumi.output(objs).apply(objs => parseYamlDocument({
                objs: objs,
                transformations: config && config.transformations || []
            }, {parent: this}));
        }
    }
```